### ISSUE TITLE
Finger print sensor not working while unlocking the device.

### ISSUE THREAD

**Issue by ASWATH-BHARATHWAJ** [phhusson/treble_experimentations#1825]
Device name: Realme Narzo 20a
issue  details:  on every gsi builds, the finger print setup on settings take the input as finger 1. But when i try to unlock using that finger there is not even a vibration that makes us understand that the Fp scanner detects and as result i have use other type of unlock like pattern or other.  

[log.txt](https://github.com/phhusson/treble_experimentations/files/6442264/log.txt)
[features.txt](https://github.com/phhusson/treble_experimentations/files/6442266/features.txt)
[lshal.txt](https://github.com/phhusson/treble_experimentations/files/6442268/lshal.txt)
[prop.txt](https://github.com/phhusson/treble_experimentations/files/6442269/prop.txt)

**Comment by ASWATH-BHARATHWAJ** [comment]
Here is the logs taken when i added finger1 input into settings >security.


[fp-input-logs-narzo20a.txt](https://github.com/phhusson/treble_experimentations/files/6442395/fp-input-logs-narzo20a.txt)

**Comment by ASWATH-BHARATHWAJ** [comment]
Now i am  adding the fp logs from stock rom..



Here is the logs taken when i added finger1 input into settings >security.
[fp stock rom input logs.txt](https://github.com/phhusson/treble_experimentations/files/6507112/fp.stock.rom.input.logs.txt)

and the second log is taken when the screen is off and while trying to unlock with fp sensor.
[while trying to unlock with fp -stock rom.txt](https://github.com/phhusson/treble_experimentations/files/6507111/while.trying.to.unlock.with.fp.-stock.rom.txt)

**Comment by phhusson** [comment]
Stock has:
```
05-19 14:21:41.181  1992  1992 V FingerprintService: startAuthentication(com.android.systemui)
05-19 14:21:41.181  1992  1992 V FingerprintService: starting client AuthenticationClientImpl(com.android.systemui) targetUserId: 0 currentUserId: 0 cookie: 0/0
05-19 14:21:41.190  1453  1672 I FingerprintHal: [egis_fingerprint.c] [fingerprint_authenticate:586] fingerprint_authenticate enter, operation_id = 0, gid = 0
```

On GSI, we get FingerprintService's strat authentication, but no answer from FingerprintHal. I'm surprised that operation_id = gid = 0 on stock
I'll decompile fingerprint HAL to know what's going on.

**Comment by ASWATH-BHARATHWAJ** [comment]
ok, sure . plz ask me if you need any files or log from stock 
thanks .

**Comment by azharmansoori22** [comment]
> Stock has:
> 
> ```
> 05-19 14:21:41.181  1992  1992 V FingerprintService: startAuthentication(com.android.systemui)
> 05-19 14:21:41.181  1992  1992 V FingerprintService: starting client AuthenticationClientImpl(com.android.systemui) targetUserId: 0 currentUserId: 0 cookie: 0/0
> 05-19 14:21:41.190  1453  1672 I FingerprintHal: [egis_fingerprint.c] [fingerprint_authenticate:586] fingerprint_authenticate enter, operation_id = 0, gid = 0
> ```
> 
> On GSI, we get FingerprintService's strat authentication, but no answer from FingerprintHal. I'm surprised that operation_id = gid = 0 on stock
> I'll decompile fingerprint HAL to know what's going on.

Yes same issue found on every phone like (realme 3 RMX1821) having egis fp. please do needful and appreciate your work. hope this will fix soon.
